system/Bootstrap.cfc

```diff
@@ -155,9 +156,9 @@ component serializable="false" accessors="true" {
 						// Remove any context stragglers
 						structDelete( request, "cb_requestContext" );
 					} catch ( any e ) {
+						// reset our application state to prevent any stale app keys from hanging around
+						applicationStop();
 						rethrow;
-					} finally {
```

Comment:
So is this the bug where ACF does NOT run any `finally` blocks?

Reply:
Yes and no. With an `applicationStop` in the catch.  You wouldn't need a `finally` anyway.  That said, I've got some tweaks to this PR.  After developing for a bit with these changes, I've found that #2 in the caveat above is a pain to work with.  Also, `applicationStop`  seems to remove the Coldbox and Wirebox keys, but keeps the old FW reinit keys.  I need to do some more work with this, before we merge this change.